### PR TITLE
Make env var optional in release-automation example

### PR DESCRIPTION
**Pull request by maackle**
### INCLUDED CHANGES:
- ibid.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info

### CODE REVIEW

File: crates/release-automation/examples/fix-json.rs [hunk 1]
```diff
@@ -27,7 +27,10 @@ fn get_clippy_output() -> Result<PathBuf, Error> {
     cmd.args(&[
         "clippy",
         "--target-dir",
-        &format!("{}/clippy", env!("CARGO_TARGET_DIR")),
+        &format!(
+            "{}/clippy",
+            option_env!("CARGO_TARGET_DIR").unwrap_or("target")
```
Comment (steveeJ) [inline crates/release-automation/examples/fix-json.rs:32]: according to https://doc.rust-lang.org/cargo/reference/environment-variables.html this env var should be set by `cargo`. i didn't envision running this without `cargo` but i'd like to learn if you are and why.

Comment (maackle) [reply crates/release-automation/examples/fix-json.rs:32]: @steveeJ the docs are misleading, it says it "sets" the env variable, but I don't think it literally does that.

Running `cargo run --example fix-json` without this change produces:

```
error: environment variable `CARGO_TARGET_DIR` not defined
  --> crates/release-automation/examples/fix-json.rs:30:31
   |
30 |         &format!("{}/clippy", env!("CARGO_TARGET_DIR")),
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Comment (steveeJ) [reply crates/release-automation/examples/fix-json.rs:32]: that's odd. is `CARGO_BUILD_TARGET_DIR` set?

Comment (maackle) [reply crates/release-automation/examples/fix-json.rs:32]: no, that's not set either.

Comment (steveeJ) [reply crates/release-automation/examples/fix-json.rs:32]: that's odd as well. ideally we'd have code that looks for the cargo workspace root to use its target directory.

**Review by steveeJ — APPROVED** [review]
since this code isn't integrated with the actual release-automation code i think we're fine to merge this as is. if we do integrate this with then we can always add what i described in the line comment.

:+1:

**Pull request merged**